config/params/values.go

```diff
@@ -18,4 +18,56 @@ var ConfigNames = map[ConfigName]string{
 }
 
 // ConfigName enum describes the type of known network in use.
-type ConfigName = int
+type ConfigName int
+
+func (n ConfigName) String() string {
+	s, ok := ConfigNames[n]
+	if !ok {
+		return "undefined"
+	}
+	return s
+}
+
+func AllConfigs() map[ConfigName]*BeaconChainConfig {
+	all := make(map[ConfigName]*BeaconChainConfig)
+	for name := range ConfigNames {
+		var cfg *BeaconChainConfig
+		switch name {
+		case Mainnet:
+			cfg = MainnetConfig()
+		case Prater:
+			cfg = PraterConfig()
+		case Pyrmont:
+			cfg = PyrmontConfig()
+		case Minimal:
+			cfg = MinimalSpecConfig()
+		case EndToEnd:
+			cfg = E2ETestConfig()
+		}
+		cfg = cfg.Copy()
+		cfg.InitializeForkSchedule()
+		all[name] = cfg
+	}
+	return all
+}
+
+type ForkName int
```

Comment:
Any chance we can use `runtime/version/fork.go` or modify that to fit your use case?

It feels like we are accomplishing the samething in two different places

Reply:
will take a look

Reply:
so this is kind of janky, because there isn't a custom type for the defs in `runtime/version/fork.go`, so there are comparisons to `int` alll over the place, meaning switching it to a custom type is a big change. I guess I can let go of the custom type and just use a plain int here =/ i'd like to follow up and make this a custom type in a subsequent PR.

Reply:
Also noting that with the `runtime/version` flavor, we have to do `version.String(v)` vs the type implementing Stringer and being usable as a string.